### PR TITLE
Metal ifmr

### PR DESCRIPTION
**Pull request by samrose30**
Add metallicity dependent options for the IFMR object in PopSyCLE

What's Changed?

- IFMR object is now a required argument for perform_pop_syn in synthetic.py
- Run.py + the documentation and example notebooks have been changed to support this change
- Currently there are two supported SPISEA IFMR objects, Spera15, and Raithel18.
- Depending on which IFMR object the user chooses the Galaxia stars are run in different metallicity bins (used to be popid[age], now it is popid[age][metallicity]. Spera15 has 4 hardcoded metallicity bins, while Raithel18 runs only at solar metallicity
- In _make_comp_dict the the SPISEA cluster has been updated to run with the specified IFMR object
- Compact objects are assigned a metallicity drawn from the distribution of the Galaxia stars from which the SPISEA cluster parameters are drawn
- Also have updated all mentions of PopStar and PyPopStar to SPISEA in all files

Potential Problems:

- Needed additional MIST isochrones to run (10.02-10.14) log_age for the non-solar metallicities
- Took a long time to generate the isochrones needed for 4 metallicities rather than just the solar ones. This is a one time cost however
- Dependent on the metal_ifmr updates to SPISEA. Need the updated versions of both for everything to work properly.

### CODE REVIEW

File: popsycle/synthetic.py [hunk 1]
```diff
@@ -1085,6 +1155,9 @@ def _make_comp_dict(iso_dir, log_age, currentClusterMass,
         The next unique ID number (int) that will be assigned to
         the new compact objects created.
 
+    unmade_cluster_counter: int
```
Comment (caseylam) [inline popsycle/synthetic.py:1158]: Can you also calculate the total mass in the unmade clusters, in addition to the number?

File: popsycle/synthetic.py [hunk 2]
```diff
@@ -1211,9 +1291,9 @@ def _make_comp_dict(iso_dir, log_age, currentClusterMass,
             comp_dict['zams_mass'] = comp_table['mass'].data
 
             ##########
-            # Assign spherical positions and velocities to all compact objects.
+            # Assign spherical positions, and velocities to all compact objects.
```
Comment (caseylam) [inline popsycle/synthetic.py:1294]: (Nitpicking, I don't think a comma is required here)

File: popsycle/synthetic.py [hunk 3]
```diff
@@ -1304,17 +1384,26 @@ def _make_comp_dict(iso_dir, log_age, currentClusterMass,
...
-            comp_dict['feh'] = np.full(len(comp_dict['vx']), np.nan)
 
             if additional_photometric_systems is not None:
                 if 'ztf' in additional_photometric_systems:
                     comp_dict['ztf_g'] = np.full(len(comp_dict['vx']), np.nan)
                     comp_dict['ztf_r'] = np.full(len(comp_dict['vx']), np.nan)
                     comp_dict['ztf_i'] = np.full(len(comp_dict['vx']), np.nan)
 
+            #########
+            # Initialize values for compact object teff, specific gravity and bolometic luminosity
```
Comment (caseylam) [inline popsycle/synthetic.py:1395]: bolometric

**Review by caseylam — COMMENTED** [review]
Looks good!

File: popsycle/synthetic.py [hunk 4]
```diff
@@ -692,163 +729,183 @@ def perform_pop_syn(ebf_file, output_root, iso_dir,
...
+                    star_dict['vz'] = ebf.read_ind(ebf_file, '/vz', bin_idx)
+                    star_dict['age'] = age_array[bin_idx]
+                    star_dict['popid'] = popid_array[bin_idx]
+                    star_dict['exbv'] = ebf.read_ind(ebf_file, '/exbv_schlegel', bin_idx)
+                    star_dict['glat'] = ebf.read_ind(ebf_file, '/glat', bin_idx)
+                    star_dict['glon'] = ebf.read_ind(ebf_file, '/glon', bin_idx)
+                    star_dict['mbol'] = ebf.read_ind(ebf_file, '/lum', bin_idx)
+                    star_dict['grav'] = ebf.read_ind(ebf_file, '/grav', bin_idx)
+                    star_dict['teff'] = ebf.read_ind(ebf_file, '/teff', bin_idx)
+                    star_dict['feh'] = ebf.read_ind(ebf_file, '/feh', bin_idx)
```
[the remark this answers is not in the document]
Comment (samrose30) [reply popsycle/synthetic.py:812]: I think we want to keep the metallicity from Galaxia for the stars. The compact objects are assigned a metallicity from this distribution.

**Pull request closed without merging**